compiler/rustc_lexer/src/unescape.rs

```diff
@@ -118,15 +118,15 @@ pub enum Mode {
     Str,
     Byte,
     ByteStr,
-    RawStr,
-    RawByteStr,
+    RawStr(u16),
+    RawByteStr(u16),
```

Comment:
What do these `u16` represent?

Reply:
Looks like the number of `#` characters used in a raw string like (`r##"hello world"##`). A doc comment would be nice.

Reply:
These numbers don't logically belong to `Mode` though.

Not sure why some of this span arithmetic is done outside of `emit_unescape_error` and then undone inside it, maybe it can be simplified, need to look in more detail.

Reply:
We need both the span for the outer span (`"..."`) and inner span (`...`). An alternative would be to pass those two spans into the function (they are both needed) to avoid carrying that information in the `Mode`.

Reply:
Changed to pass in both `Span`s (as both are needed for different errors).